### PR TITLE
[BACKLOG-36841]-Accessibility-PUC Home Page – Home Perspective -Getting Started Section.

### PR DESCRIPTION
**Pull request by phalguni-a**

### CODE REVIEW

File: user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js [hunk 1]
```diff
@@ -85,6 +85,32 @@ define([
...
+          if (firstButtonId === this.id) {
+            $('#btnCreateNew').popover('hide');
+            $('#btnCreateNew').focus();
+            event.preventDefault();
+            event.stopPropagation();
+          }
+        } else if (event.keyCode === 27) {
+          $('#btnCreateNew').popover('hide');
+          $('#btnCreateNew').focus();
+        } else if (event.keyCode === 9) {
```
Comment (NJtwentyone) [inline user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js:102]: I'm not too familiar with javascript coding standards. I'll accept push back on this...

all the `event.keyCode === X`  have magic numbers: https://codehs.com/glossary/term/14

I would either declare a constant or write a comment by each one stating the meaning of each number.

1. it easier to quickly read without referencing external documentation.
2. based off Mozilla https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#constants_for_keycode_value this feature might be deprecated, so better to document what the original author intended as well. Some browsers might not break compatibility in the future.

Comment (phalguni-a) [reply user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js]: I agree, its meaningful to add what that number means as well. I added the comments as you suggested.

File: user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js [hunk 2]
```diff
@@ -85,6 +85,32 @@ define([
...
+      $('.popover .btn').keydown(function (event) {
+        if (event.shiftKey && event.keyCode === 9) {
+          var firstButtonId = $('.popover .btn').first().attr("id");
+          if (firstButtonId === this.id) {
+            $('#btnCreateNew').popover('hide');
+            $('#btnCreateNew').focus();
+            event.preventDefault();
+            event.stopPropagation();
+          }
+        } else if (event.keyCode === 27) {
```
[the remark this answers is not in the document]
Comment (phalguni-a) [reply user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js]: https://stackoverflow.com/questions/4471582/keycode-vs-which
Looks like its better to support event.which as well.
Did the changes for the same.

[the remark this answers is not in the document]
Comment (NJtwentyone) [reply user-console/src/main/resources/org/pentaho/mantle/public/home/js/home.js]: no further action here.

Just for future reference.
based off your stackoverflow link and the mozilla we should actually be using "event.key" https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key. _**But I don't think all modern browsers and/or non desktop browsers support**_ it https://www.w3.org/TR/uievents/#dom-keyboardevent-code

**Review by NJtwentyone — APPROVED** [review]

**Review by rmansoor — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-11/job/PR-5193/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 20m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5193/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**